### PR TITLE
[MM-42068] Adding source query param to team and channel group endpoints

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
Adding `group_source` to the query parameters in order to filter out custom groups

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42068
https://mattermost.atlassian.net/browse/MM-42070

#### Release Note

```release-note
Fixes a bug where custom groups were appearing in the UI when linking groups to channels/teams
```

### CODE REVIEW

**Review by mkraft — APPROVED** [review]

**Review by mylonsuren — COMMENTED** [review]
Should a test case be added to verify that custom groups are no longer included in the response?

**Comment by BenCookie95** [comment]
> Should a test case be added to verify that custom groups are no longer included in the response?

There's a test on server to verify that the endpoint returns groups based on that query param, so I think we're good

**Review by mylonsuren — APPROVED** [review]

**Comment by BenCookie95** [comment]
Discovered an issue so need to add another part to this

**Comment by BenCookie95** [comment]
@mkraft & @mylonsuren This needs a re-review, had to adjust the redux selectors

File: packages/mattermost-redux/src/selectors/entities/groups.ts [hunk 1]
```diff
@@ -148,7 +150,7 @@ export const getGroupsNotAssociatedToTeam: (state: GlobalState, teamID: string)
     getAllGroups,
     (state: GlobalState, teamID: string) => getTeamGroupIDSet(state, teamID),
     (allGroups, teamGroupIDSet) => {
-        return Object.entries(allGroups).filter(([groupID]) => !teamGroupIDSet.has(groupID)).map((entry) => entry[1]);
+        return Object.entries(allGroups).filter(([groupID]) => !teamGroupIDSet.has(groupID)).filter((entry) => (entry[1].source === Constants.LDAP_SERVICE)).map((entry) => entry[1]);
```
Comment (mylonsuren) [inline packages/mattermost-redux/src/selectors/entities/groups.ts:153]: Could we do something similar to ln. 173 in `getGroupsNotAssociatedToChannel` and removed the extra filter here?

```suggestion
        return Object.entries(allGroups).filter(([groupID, group]) => !teamGroupIDSet.has(groupID) && group.source === Constants.LDAP_SERVICE).map((entry) => entry[1]);
```

**Review by mylonsuren — CHANGES_REQUESTED** [review]

File: packages/mattermost-redux/src/selectors/entities/groups.ts (hunk 1 above)
Comment (BenCookie95) [reply packages/mattermost-redux/src/selectors/entities/groups.ts:153]: Yeah good suggestion

**Review by mkraft — APPROVED** [review]

**Review by mylonsuren — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
Working as expected

**Before**

image

**After**

image

**Pull request merged**